javascript/javascript1/week4/voice_assistant.js

```diff
@@ -0,0 +1,124 @@
+'use strict'
+let name = '';
+let todoList = [];
+let map = [
+    { country: 'Denmark', capital: 'Copenhagen' },
+    { country: 'Ukraine', capital: 'Kyiv' },
+    { country: 'Italy', capital: 'Rome' },
+    { country: 'Poland', capital: 'Warsaw' },
+    { country: 'Belarus', capital: 'Minsk' },
+    { country: 'Turkey', capital: 'Istambul' },
+]
+
+function getReply(command) {
+    if (command.startsWith('Hello my name is')) {
+        const startIndex = command.lastIndexOf(' ');
+        const newName = command.slice(startIndex + 1, command.length);
+        if (name === newName) {
+            return 'We are already friends, ' + name;
+        } else {
+            name = newName;
+            return 'Nice to meet you ' + name;
+        }
+    } else if (command === 'What is my name') {
+        if (name === '') {
+            return "I don't know, we haven't met yet";
+        } else {
+            return `Your name is: ${name}`;
+        }
+    } else if (command.startsWith('Add') && command.endsWith('to my todo')) {
+        const todoStartIndex = command.indexOf(' ');
+        const todoEndIndex = command.indexOf('to my todo');
+        const action = command.slice(todoStartIndex, todoEndIndex).trim();
+        todoList.push(action);
+        return `${action} added to your todo`;
+    } else if (command.startsWith('Remove') && command.endsWith('from my todo')) {
+        const todoStartIndex = command.indexOf(' ');
+        const todoEndIndex = command.indexOf('from my todo');
+        const action = command.slice(todoStartIndex, todoEndIndex).trim();
+        for (let i = 0; i < todoList.length; i++) {
+            if (action === todoList[i]) {
+                todoList.splice(i, 1);
+                return `Removed ${action} from your todo`;
+            }
+        }
+        return `${action} you want to remove is not on your list`;
+    } else if (command === 'What is on my todo?') {
+        const list = todoList.join(', ');
+        return `You have ${todoList.length} todos - ${list}`;
+    } else if (command === 'What day is it today?') {
+        const date = new Date();
+        const p = date.toDateString();
+        return `Today is ${p}`;
+        // Extra command: countries and their capitals
+    } else if (command.startsWith('What is the capital of ')) {
+        const startIndex = 'What is the capital of '.length;
+        const endIndex = command.indexOf('?');
+        const country = command.slice(startIndex, endIndex);
+        for (let i = 0; i < map.length; i++) {
+            if (map[i].country === country) {
+               return `The capital of ${map[i].country} is ${map[i].capital}`;
+            }
+        }
+        return `I don't know what is the capital of ${country}`;
+    } else if (command.startsWith('what is')) {
+        const startIndex = 'What is '.length;
+        const lastIndex = command.length;
+        const evaluation = command.slice(startIndex, lastIndex);
+        return eval(evaluation);
```

Comment:
eval is not recommended.
Check [https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval](url)
You can also change command to array of words then get numbers and operation like this for example 

 const commandWords = command.split(" ");
    const number1 = parseInt(commandWords[2]);
    const number2 = parseInt(commandWords[4]);
    const operation = commandWords[3];
    if (number1 && number2) {
      switch (operation) {
        case "+":
          return number1 + number2;
        case "-":
          return number1 - number2;
        case "*":
          return number1 * number2;

Reply:
I changed the file. Thank you for your feedback :)